app/modules/platform/saga/api/apiPostSigninToTokenAndGetUserAuth.js

```diff
@@ -23,24 +24,23 @@ const apiPostSigninToTokenAndGetUserAuth = function* (
   try {
     const responseData: ApiResponseData = yield call(api.token.postSignin, email, password);
 
-    // Get flow to stop complaining about token possibly being NULL
-    if (responseData.token == null) {
-      throw new Error(`This shouldn't happen`);
+    if (responseData.token == null || responseData.body == null) {
+      throw new UnexpectedEmptyResponseError();
```

Comment:
The error still needs renaming to something that encompasses both of these issues. `UnexpectedHttpResponseError` or `InvalidHttpResponseError` or whatever you think is appropriate.